core/src/main/scala/com/nvidia/spark/rapids/tool/analysis/AppSQLPlanAnalyzer.scala

```diff
@@ -77,100 +189,28 @@ class AppSQLPlanAnalyzer(app: AppBase, appIndex: Int) extends AppAnalysisBase(ap
     }
     connectOperatorToStage(createGraphFunc)
     for (sqlPIGEntry <- sqlPlanInfoBuffer.sqlPlanInfoGraphs) {
-      var sqlIsDsOrRDD = false
-      val potentialProblems = collection.mutable.Set[String]()
-      // store all datasources of the given SQL in a variable so that we won't have to iterate
+      // store all dataSources of the given SQL in a variable so that we won't have to iterate
       // through the entire list
       // get V1 dataSources for that SQLId
-      val sqlDataSources = app.checkMetadataForReadSchema(sqlPIGEntry)
+      val visitorContext = SQLPlanVisitorContext(sqlPIGEntry,
```

Comment:
I'm a little unclear why we need this mutable visitorContext at all. Can't we just do the changes in the visit node function?

Reply:
I needed to define method VisitNode that allows to use the same main logic  between Qual/Profiler.
The mutable visitor has some data structured and flags that are updated each time a node is visited.
Finally, after visiting all the nodes, the final values can be set for the SQLPlan.
`VisitNode` made it feasible for Qualification tool to extend that class and override the logic to update the `app.writeDataFormat`
In future code improvements, we can have better asbstraction and more types/customizations